vendor/parser-extended/parsetree.mli

```diff
@@ -470,6 +471,40 @@ and expression_desc =
   | Pexp_indexop_access of indexop_access
   | Pexp_prefix of string loc * expression  (** [op E] *)
   | Pexp_infix of string loc * expression * expression  (** [E1 op E2] *)
+  | Pexp_list_comprehension of comprehension
+      (** [[BODY ...CLAUSES...]] *)
+  | Pexp_array_comprehension of mutable_flag * comprehension
+      (** [[|BODY ...CLAUSES...|]] (flag = Mutable)
+          [[:BODY ...CLAUSES...:]] (flag = Immutable)
+          (only allowed with [-extension immutable_arrays]) *)
+  | Pexp_immutable_array of expression list  (** [[: E1; ...; En :]] *)
```

Comment:
I think that, on balance, it would be simpler to add a new field to the existing `Pexp_array`. I might be wrong, but that's my hunch.

Reply:
done